src/PureScript/CST/Utils/ModuleGraph.purs

```diff
@@ -0,0 +1,87 @@
+module PureScript.CST.Utils.ModuleGraph
+  ( moduleGraph
+  , sortModules
+  ) where
+
+import Prelude
+
+import Control.Monad.Free (Free, runFree)
+import Data.Array as Array
+import Data.Foldable (all, foldl)
+import Data.Identity (Identity(..))
+import Data.Map (Map)
+import Data.Map as Map
+import Data.Maybe (Maybe(..), fromMaybe, maybe)
+import Data.Newtype (un)
+import Data.Set (Set)
+import Data.Set as Set
+import Data.Traversable (for)
+import Data.Tuple (Tuple(..))
+import PureScript.CST.Types (ImportDecl(..), Module(..), ModuleHeader(..), ModuleName, Name(..))
+
+type Graph a = Map a (Set a)
+
+moduleGraph :: forall e. Array (Module e) -> Graph ModuleName
+moduleGraph = Map.fromFoldable <<< map go
+  where
+  go (Module { header: ModuleHeader { name: Name { name }, imports } }) =
+    Tuple name (Set.fromFoldable (map getImportName imports))
+
+  getImportName (ImportDecl { module: Name { name } }) = name
```

Comment:
We should quote this for now since the highlighter hates it.

Reply:
Sure